packages/nft/src/NFTHandler.ts

```diff
@@ -240,4 +234,65 @@ export class NFTHandler {
     }
     return tokenUri
   }
+
+  /**
+   * It takes an address and returns its balance
+   * @param {NFT721Manager} nftManager - NFT721Manager - The NFT721Manager instance that will be used
+   * to fetch the balance.
+   * @param {string} address - The address of which we want the balance.
+   * @returns The balance
+   */
+  @ManagerMustBeActive()
+  private static async getAddressBalance(
+    nftManager: NFT721Manager,
+    address: string
+  ): Promise<number> {
+    if (!nftManager.implements('balanceOf', ['address'])) {
+      throw FUNCTION_NOT_IMPLEMENTED(
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        nftManager.contractAddr!,
+        'balanceOf(address)'
+      )
+    }
+
+    try {
+      const { value } = await nftManager.execute({
+        method: 'balanceOf(address)',
+        params: [address]
+      })
+      return value
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    } catch (error: any) {
+      throw BALANCE_OF_FAILED(address, error)
+    }
+  }
```

Comment:
is the try/catch needed?

Reply:
It depends, to give a little clearer error to the dev.